### DISCUSSION TITLE
Does MirrorD actually replace the interpreter of the running process?

### DISCUSSION THREAD

**Discussion by tomp332** [metalbear-co/mirrord#2174] [Q&A]
I am using MirrorD for a Python based Deployment development.
I have a few questions regarding the architecture magic that is happening: 
1. Does MirrorD replace completely the Python process that is running in the remote deployment? Meaning that if I install local Python packages that do not exist in the remote Python pod, will they entire Python process just be applied?
2. Is there a way to make MirrorD an instellisence\ interpreter for non run time process? Meaning for it to be available for static checks and code interpretation while developing? 

Thanks alot

**Comment by aviramha** [comment]
hey @tomp332 

1. No, it doesn't replace nor it uses it's packages - mirrord runs your process locally, providing file access to the remote based on filters. By default, most files that are code-related (packages, libraries, binaries..) are loaded locally and don't change the remote service. mirrord does enable you to steal traffic from the remote pod, meaning that your local process will be the one answering for the incoming requests.
2. I'm not sure I understand that question but maybe you're asking if you could have code completion based on the remote environment's code? if yes, no we don't support that and it's not the way it works in run time as well.
